Makefile

```diff
@@ -145,8 +145,8 @@ define cmake-build
 +@$(eval PX4_CONFIG = $(1))
 +@$(eval BUILD_DIR = $(SRC_DIR)/build/$(PX4_CONFIG)$(BUILD_DIR_SUFFIX))
 +@if [ $(PX4_CMAKE_GENERATOR) = "Ninja" ] && [ -e $(BUILD_DIR)/Makefile ]; then rm -rf $(BUILD_DIR); fi
-+@if [ ! -e $(BUILD_DIR)/CMakeCache.txt ]; then mkdir -p $(BUILD_DIR) && cd $(BUILD_DIR) && cmake $(SRC_DIR) -G"$(PX4_CMAKE_GENERATOR)" $(CMAKE_ARGS) -DCONFIG=$(PX4_CONFIG) || (rm -rf $(BUILD_DIR)); fi
-+@$(PX4_MAKE) -C $(BUILD_DIR) $(PX4_MAKE_ARGS) $(ARGS)
++@if [ ! -e $(BUILD_DIR)/CMakeCache.txt ]; then mkdir -p $(BUILD_DIR) && cd $(BUILD_DIR) && CCACHE_BASEDIR=$(BUILD_DIR) cmake $(SRC_DIR) -G"$(PX4_CMAKE_GENERATOR)" $(CMAKE_ARGS) -DCONFIG=$(PX4_CONFIG) || (rm -rf $(BUILD_DIR)); fi
++@CCACHE_BASEDIR=$(BUILD_DIR) $(PX4_MAKE) -C $(BUILD_DIR) $(PX4_MAKE_ARGS) $(ARGS)
```

Comment:
I don't think we should handle CCACHE_BASEDIR here. You'll get inconsistent behavior if moving between an IDE and building on the command line (a use case supported with our vscode config).

I was thinking we could push it into cmake, but there's not really a benefit to doing this locally.

How about setting it in the appveyor.yml? This is what Jenkins does.
https://github.com/PX4/Firmware/blob/a3ebe5676f04c546b999d21b53343ee56a2f92a6/Jenkinsfile#L15